File: CPlusPlus/src/Lion.cpp
#include <iostream>
#include <string>
#include "../include/Lion.h"

namespace ClassZoo {
ClassZoo::Lion::Lion(std::string inputName, int inputID,
    float inputWeight, float inputHeight)
    : ClassZoo::Animal(inputName, inputID, inputWeight, inputHeight) {}
void ClassZoo::Lion::PrintSpecies() {
    std::cout << "Animal Species: Lion\n";
}
void ClassZoo::Lion::AnimalSound() {
    std::cout << ClassZoo::Animal::GetName() << " roars\n";
}
}

File: Java/src/classZoo/Lion.java
package classZoo;

class Lion extends Animal {

	public Lion(String inputName, int inputID, float inputWeight, float inputHeight) {
		super(inputName, inputID, inputWeight, inputHeight);
	}
	
	public void printSpecies() {
		System.out.println("Animal Species: Lion");
	}
	
	public void animalSound() {
		System.out.printf("%s roars\n", this.getName());
	}

}
